### PR TITLE
Use ember-remodal to simplify modal use

### PR DESCRIPTION
**Pull request by rox163**
# major
# CHANGELOG
## Changed
- BREAKING Using [ember-remodal](http://sethbrasile.github.io/ember-remodal/) addon to simplify modal dialog support. Additional modal-dialog template options required
## Removed
- BREAKING Removed liquid-fire modal code. 
  - Removed the need to configure the router.js file
## Upgrade notes

If you are upgrading your app to use this version, note the following - 
- Remove the modal declaration from your router.js file
- Remove `{{liquid-modal}}` template code if you do not have any other liquid-fire modals in your app
- Apply `remodal-bg` class to the parent container where you want a blur effect on the modal overlay.
- `frost-modal-dialog` root template must use block-slots with `target` and `body` slots. See README for more details.

### CODE REVIEW

**Comment by rox163** [comment]
This PR will resolve #4

File: addon/components/frost-modal-dialog.js [hunk 1]
```diff
@@ -28,13 +28,36 @@ export default Component.extend({
...
-    cancel () {
-      this.sendAction('dismiss')
-    },
     confirm () {
-      this.sendAction('confirm')
-      this.sendAction('dismiss')
+      this.get('onConfirmHandler')()
+    },
+    modalOpen () {
+      // this.set('$containerEl', Ember.$('.ember-remodal.window.remodal-is-opened .ps-container'))
```
Comment (rox163) [inline addon/components/frost-modal-dialog.js:40]: depending on UX feedback we may use the same scrolling setup+style as used in frost-modal-input

File: tests/integration/components/frost-modal-dialog-test.js [hunk 2]
```diff
@@ -15,37 +14,37 @@ describeComponent(
...
-      this.set('clicked', false)
-      this.on('confirm', function () {
-        this.set('clicked', true)
-      })
-
-      this.render(hbs`{{frost-modal-dialog type='confirmation'
-        title="Confirmation" confirmAlias="Confirm" message="Confirmation message" confirm=(action 'confirm')}}`)
-      Ember.run(() => this.$('.frost-button').get(1).click())
-      assert.isTrue(this.get('clicked'), 'confirmed')
+    //   this.render(hbs`{{frost-modal-dialog type='information'
```
Comment (rox163) [inline tests/integration/components/frost-modal-dialog-test.js]: Cant get block-slots working in these tests yet :(

File: addon/styles/addon.scss [hunk 3]
```diff
@@ -1,60 +1,59 @@
...
-      line-height: 30px;
+          .frost-button {
+            width: 150px;
+            margin-right: 10px;
+          }
+        }
+      }
     }
   }
+
```
Comment (juwara0) [inline addon/styles/addon.scss]: There are two empty lines here

File: tests/integration/components/frost-modal-dialog-test.js (hunk 2 above)
Comment (rox163) [inline tests/integration/components/frost-modal-dialog-test.js]: need to push an update. i got the dialog test working..

File: addon/templates/components/frost-modal-dialog.hbs [hunk 4]
```diff
@@ -1,44 +1,64 @@
...
+  forService=true
+  closeOnOutsideClick=false
+  onOpen=(action 'modalOpen')
+  onClose=(action 'modalClose')
+  onConfirm=(action 'confirm')
+  closeOnEscape=false
+  disableNativeClose=true as |modal|
+}}
+  {{yield}}
+  {{#modal.open}}
```
Comment (juwara0) [inline addon/templates/components/frost-modal-dialog.hbs:11]: Is the intent to always have {{modal.open}} show or would you want this to only show if the block-slot 'target' is used? So should the {{modal.open}} be wrapped inside the {{yield-slot 'target'}} ?

File: addon/templates/components/frost-modal-dialog.hbs [hunk 5]
```diff
@@ -1,44 +1,64 @@
...
       {{/if}}
     </div>
+  {{/yield-slot}}
+
+  <div class="footer">
+    {{#if confirmAlias}}
+      {{#modal.cancel}}
+        {{frost-button
+          autofocus=true
+          size="medium"
```
Comment (juwara0) [inline addon/templates/components/frost-modal-dialog.hbs]: Are you wanting to use single quotes here for the attribute? size='medium'

I saw other instances for this below as well.

File: addon/templates/components/frost-modal-dialog.hbs (hunk 4 above)
Comment (rox163) [inline addon/templates/components/frost-modal-dialog.hbs:11]: The yield wouldnt work properly if it was wrapping the modal.open block for some reason. Also if the block isnt defined in the template, nothing shows anyway. The #modal.open is just a block that connects whatever it contains to the onOpen action on click.

**Comment by juwara0** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-modal-dialog/pull-request/3/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-modal-dialog)

**Pull request merged**